pkg/addons/addon_ufw.go

```diff
@@ -0,0 +1,89 @@
+package addons
+
+import (
+	"fmt"
+
+	"github.com/xetys/hetzner-kube/pkg/clustermanager"
+)
+
+//UfwAddon installs ufw
+type UfwAddon struct {
+	masterNode   *clustermanager.Node
+	communicator clustermanager.NodeCommunicator
+	nodeCidr     string
+	nodes        []clustermanager.Node
+}
+
+//NewUfwAddon installs ufw to the cluster
+func NewUfwAddon(provider clustermanager.ClusterProvider, communicator clustermanager.NodeCommunicator) ClusterAddon {
+	masterNode, _ := provider.GetMasterNode()
+	return UfwAddon{masterNode: masterNode, communicator: communicator, nodeCidr: provider.GetNodeCidr(), nodes: provider.GetAllNodes()}
+}
+
+func init() {
+	addAddon(NewUfwAddon)
+}
+
+//Name returns the addons name
+func (addon UfwAddon) Name() string {
+	return "ufw"
+}
+
+//Requires returns a slice with the name of required addons
+func (addon UfwAddon) Requires() []string {
+	return []string{}
+}
+
+//Description returns the addons description
+func (addon UfwAddon) Description() string {
+	return "Uncomplicated Firewall"
+}
+
+//URL returns the URL of the addons underlying project
+func (addon UfwAddon) URL() string {
+	return "https://wiki.ubuntu.com/UncomplicatedFirewall"
+}
+
+//Install performs all steps to install the addon
+func (addon UfwAddon) Install(args ...string) {
+
+	var nodeIpRules string
+	for _, node := range addon.nodes {
+		nodeIpRules += " && ufw allow in from " + node.IPAddress + " to any"
+	}
+	var output string
+	for _, node := range addon.nodes {
```

Comment:
We apply the same rule for each node, but I think each kind of node should have some specific rules, eg: etcd/master node don't need to expose 80/443 (that I suppose should be exposed on workers); maybe I'm missing some part of the setup can you better explain me what we like to achieve with this addon?

Reply:
So I think the ufw rules are the trickiest part of this addon. I tried to use the most common rules any new cluster would require. But you can always change these rules via script runner from #229 for ex.